src/connectRoutes.js

```diff
@@ -126,6 +130,14 @@ export default (
     initialDispatch: shouldPerformInitialDispatch = true
   }: Options = options
 
+  const selectLocationState = typeof locationKey === 'function'
+    ? locationKey
+    : state => state[locationKey]
+
+  const selectTitleState = typeof titleKey === 'function'
+    ? titleKey
+    : state => state[titleKey]
+
```

Comment:
mainly by the way i can foresee a potential future where the number of options gets out of hand, and im trying to avoid that. less API surface area to think about.